### PR TITLE
Remove use of DefinePlugin, switch to VUE_APP env vars

### PR DESCRIPTION
**Pull request by AlmightyYakob**
This uses a more standard way of injecting env vars, and removes most of the complexity from `vue.config.js`. I've also created a `.env.default` file, as a template that describes the new env vars. 

Let me know what you guys think.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "test": "tape -r esm test/**/*.test.js | tap-spec"
   },
   "dependencies": {
+    "@types/node": "^13.7.7",
```
Comment (waxlamp) [inline client/package.json]: Why is this needed?

Comment (AlmightyYakob) [reply client/package.json]: I was getting an error when referencing `process.env` saying that process wasn't defined, so this is the fix.

Comment (AlmightyYakob) [reply client/package.json]: Actually I can't reproduce that error, it seems to be working if I remove that dependency (and do a fresh install). So I can remove this.

Comment (waxlamp) [reply client/package.json]: Looking at how Vue accomplishes this injection (and using `vue inspect` to look at the DefinePlugin section), it does seem like you shouldn't need to bring in any external support for this. Essentially, Webpack is injecting a `process.env` variable into the global scope, and so it should "just work" (though, I'm not entirely getting how TypeScript is ok with this, without some kind of `declare` statement :thinking:).

File: client/src/components/AboutDialog.vue [hunk 2]
```diff
@@ -70,8 +70,7 @@
 import Vue from 'vue';
 
 import AExt from '@/components/AExt.vue';
-
-const GIT_SHA: string = process.env.VUE_APP_GIT_SHA || '';
+import { gitSha as GIT_SHA } from '@/environment';
```
Comment (waxlamp) [inline client/src/components/AboutDialog.vue]: No need to shift it to `GIT_SHA` (for consistency with the other uses).

Comment (AlmightyYakob) [reply client/src/components/AboutDialog.vue]: This is done to follow linting guidelines (and I only noticed it from the linter), as without this renaming, the `gitSha` computed prop shadows the original `gitSha` import.

**Comment by JackWilb** [comment]
Looks like the client comes up during testing and on my local. However, running `yarn test:client:up` on my local fails because of one command, `curl -s -I --max-time 0.5 http://localhost:8080/api/workspaces` 404s.

File: client/src/components/AboutDialog.vue (hunk 2 above)
Comment (waxlamp) [reply client/src/components/AboutDialog.vue]: Ah I see. This seems to be implemented poorly (my fault). I will open an issue to fix this (by passing in the SHA value as a prop).

**Comment by waxlamp** [comment]
> Looks like the client comes up during testing and on my local. However, running `yarn test:client:up` on my local fails because of one command, `curl -s -I --max-time 0.5 http://localhost:8080/api/workspaces` 404s.

Ah I figured it out: the point of this PR is that it detaches the server from the client altogether. So it's no longer viable to use localhost:8080/api to probe whether the client build has succeeded; however, we can do the same thing by just curl'ing localhost:8080 instead. See 8e3d7a5b6b1bf44e1a4e1e786e804c44f738ccbd.
